addons/hw_drivers/tools/helpers.py

```diff
@@ -134,33 +141,43 @@ def check_git_branch():
     urllib3.disable_warnings()
     http = urllib3.PoolManager(cert_reqs='CERT_NONE')
     try:
-        response = http.request('POST',
+        response = http.request(
+            'POST',
             server + "/web/webclient/version_info",
             body='{}',
-            headers={'Content-type': 'application/json'}
+            headers={'Content-type': 'application/json'},
         )
 
         if response.status == 200:
             git = ['git', '--work-tree=/home/pi/odoo/', '--git-dir=/home/pi/odoo/.git']
 
             db_branch = json.loads(response.data)['result']['server_serie'].replace('~', '-')
-            if not subprocess.check_output(git + ['ls-remote', 'origin', db_branch]):
-                db_branch = 'master'
```

Comment:
We need to keep this test: this test is necessary to be able to connect a db in master, otherwise when associating a db in master we get something like "17.5" and we cannot checkout a "17.5" branch since it doesnt exist